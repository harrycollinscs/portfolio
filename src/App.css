@font-face {
  font-family: appFont;
  src: url("../src/assets/fonts/QuincyRegular.otf") format("opentype");
}

@font-face {
  font-family: "DM Serif Display", serif !important;
  src: url("../src/assets/fonts/QuincyRegular.woff") format("woff");
  font-weight: normal;
  font-style: normal;
  font-display: swap;
}

html {
  z-index: -10;
  background: white;
  -webkit-background-size: cover;
  -moz-background-size: cover;
  -o-background-size: cover;
  background-size: cover;
  background-attachment: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
}

.App {
  text-align: center;
  font-family: "DM Serif Display", serif !important;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
