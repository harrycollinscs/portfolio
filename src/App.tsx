import React from 'react'
import './App.css';
import { useState, ReactNode } from 'react';
import styled, { css } from 'styled-components';
import Resume from './components/templates/Resume';
import About from './components/templates/About';
import All from './components/templates/All';
import Projects from './components/templates/Projects';

const Body = styled.div`
`

const TabsContainer = styled.div`
  justify-content: center;
  position: sticky;
  top: 0px;
  display: flex;
  z-index: 1000;
  padding: 10px;
  box-shadow: 0rem 0.1rem 0.3rem rgba(0,0,0,0.1);
  background-color: white;
`

const Tab = styled.div`
  padding: 4px;
  border-radius: 8px;
  text-align: center;
  cursor: pointer;
  padding: 12px 20px;
  margin: 0 4px;
  font-size: 16px;
  letter-spacing: 0.5px;
  transition: all 0.5s ease;
  user-select: none;
  color: grey;
  ${({isActive}: {isActive: boolean}) => 
    isActive ? 
    css`
      background-color: #ede0ff;
      color: #6603fc;
    ` : 
    css`
      &:hover {
        color: black;
      }
    `
  }
  
`

const TabContent = styled.div`
  height: 100%;
  align-items: center;
  justify-content: center;
`

interface TabManagerProps {
  tabs: {
    title: string
    component: ReactNode
  }[]
}

const Tabs = ({ tabs }: TabManagerProps) => {
  const [activeTab, setActiveTab] = useState(0)

  return (
    <div style={{flexDirection: 'column', display: 'flex', width: '100%'}}>
      <TabsContainer>
          {tabs.map(({title}, index: number) => (
            <Tab key={index} isActive={index===activeTab} onClick={() => setActiveTab(index)}>
              {title}
            </Tab>
          ))}
      </TabsContainer>
      <TabContent>
        {tabs[activeTab].component}
      </TabContent>
    </div>
  );
  
}

const App = () => {

  const tabs = [
    { component: <All />, title: 'All'},
    { component: <Resume />, title: 'Resume'},
    { component: <About />, title: 'About'},
    { component: <Projects />, title: 'Projects'}
  ];

  return (
    <Body>
      <Tabs tabs={tabs}/>
    </Body>
  );
}



export default App;
