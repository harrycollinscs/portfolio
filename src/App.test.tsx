import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'
import App from './App';

describe('App', () => {
  it('renders learn react link', () => {
    render(<App />);
    const linkElement = screen.getByText(/All/i);
    expect(linkElement).toBeInTheDocument();
  });
})

