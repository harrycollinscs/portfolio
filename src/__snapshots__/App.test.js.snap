// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-idXgbr kZTxrS"
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "width": "100%",
      }
    }
  >
    <div
      className="sc-dIfARi hWwhyK"
    >
      <div
        className="sc-hHTYSt YAeea"
      >
        <div
          className="sc-dmctIk nIkVC"
          onClick={[Function]}
        >
          All
        </div>
        <div
          className="sc-dmctIk gDbfHq"
          onClick={[Function]}
        >
          Resume
        </div>
        <div
          className="sc-dmctIk gDbfHq"
          onClick={[Function]}
        >
          About
        </div>
      </div>
    </div>
    <div
      className="sc-kgTSHT eCZZmj"
    >
      <div
        className="sc-bcXHqe eimxqs"
      >
        <div
          className="sc-gswNZR iuRrXo"
        >
          <div
            className="sc-dkrFOg krVsfh"
          >
            <div
              aria-roledescription="carousel"
              dir="ltr"
            >
              <div
                className="react-slideshow-container"
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
                onMouseOver={[Function]}
                onMouseUp={[Function]}
                onTouchCancel={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
              >
                <div
                  className="react-slideshow-wrapper slide "
                >
                  <div
                    className="images-wrap"
                    style={
                      Object {
                        "transform": "translate(-0px)",
                      }
                    }
                  >
                    <div
                      aria-hidden="true"
                      aria-roledescription="slide"
                      data-index={-1}
                    >
                      <div
                        className="each-slide"
                      >
                        <img
                          alt="sliding"
                          className="sc-bqWxrE lcrvEY"
                          src={Object {}}
                        />
                      </div>
                    </div>
                    <div
                      aria-hidden="false"
                      aria-roledescription="slide"
                      className="active"
                      data-index={0}
                    >
                      <div
                        className="each-slide"
                      >
                        <img
                          alt="sliding"
                          className="sc-bqWxrE lcrvEY"
                          src={Object {}}
                        />
                      </div>
                    </div>
                    <div
                      aria-hidden="true"
                      aria-roledescription="slide"
                      className=""
                      data-index={1}
                    >
                      <div
                        className="each-slide"
                      >
                        <img
                          alt="sliding"
                          className="sc-bqWxrE lcrvEY"
                          src={Object {}}
                        />
                      </div>
                    </div>
                    <div
                      aria-hidden="true"
                      aria-roledescription="slide"
                      className=""
                      data-index={2}
                    >
                      <div
                        className="each-slide"
                      >
                        <img
                          alt="sliding"
                          className="sc-bqWxrE lcrvEY"
                          src={Object {}}
                        />
                      </div>
                    </div>
                    <div
                      aria-hidden="true"
                      aria-roledescription="slide"
                      data-index={3}
                    >
                      <div
                        className="each-slide"
                      >
                        <img
                          alt="sliding"
                          className="sc-bqWxrE lcrvEY"
                          src={Object {}}
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="sc-dkrFOg xDTza"
          >
            <div
              className="sc-ipEyDJ fZqabw"
              color="#bfbfbf"
            >
              <h1>
                HARRY COLLINS
              </h1>
            </div>
          </div>
          <div
            className="sc-dkrFOg gAcZeq"
          >
            <div
              className="sc-iBYQkv bHzXBj"
            >
              <h2
                className="sc-eDvSVe lnaoEU"
              />
              <p
                className="sc-jSUZER eDYipk"
              />
              <small
                className="sc-gKPRtg hQhJbe"
              />
              <div
                className="sc-kDvujY jLhsdD"
              >
                <strong />
                <div
                  className="sc-pyfCe jYvaer"
                >
                  <p
                    className="sc-jrcTuL bBXktZ"
                    style={
                      Object {
                        "margin": 0,
                        "marginTop": "5px",
                      }
                    }
                  >
                    Hey, I'm Harry. I'm a London based Software Engineer 
                  </p>
                </div>
              </div>
              <div
                className="sc-kDvujY gYuGEa"
              >
                <strong>
                  Age
                </strong>
                <div
                  className="sc-pyfCe jYvaer"
                >
                  <p
                    className="sc-jrcTuL bBXktZ"
                    style={
                      Object {
                        "margin": 0,
                        "marginTop": "5px",
                      }
                    }
                  >
                    24
                  </p>
                </div>
              </div>
              <div
                className="sc-kDvujY gYuGEa"
              >
                <strong>
                  Location
                </strong>
                <div
                  className="sc-pyfCe jYvaer"
                >
                  <p
                    className="sc-jrcTuL bBXktZ"
                    style={
                      Object {
                        "margin": 0,
                        "marginTop": "5px",
                      }
                    }
                  >
                    London, UK
                  </p>
                </div>
              </div>
              <div
                className="sc-kDvujY gYuGEa"
              >
                <strong>
                  Interests
                </strong>
                <div
                  className="sc-pyfCe jYvaer"
                >
                  <p
                    className="sc-jrcTuL bBXktZ"
                    style={
                      Object {
                        "margin": 0,
                        "marginTop": "5px",
                      }
                    }
                  >
                    Music
                  </p>
                  <p
                    className="sc-jrcTuL bBXktZ"
                    style={
                      Object {
                        "margin": 0,
                        "marginTop": "5px",
                      }
                    }
                  >
                    Basketball
                  </p>
                  <p
                    className="sc-jrcTuL bBXktZ"
                    style={
                      Object {
                        "margin": 0,
                        "marginTop": "5px",
                      }
                    }
                  >
                    Travelling
                  </p>
                  <p
                    className="sc-jrcTuL bBXktZ"
                    style={
                      Object {
                        "margin": 0,
                        "marginTop": "5px",
                      }
                    }
                  >
                    Fitness
                  </p>
                  <p
                    className="sc-jrcTuL bBXktZ"
                    style={
                      Object {
                        "margin": 0,
                        "marginTop": "5px",
                      }
                    }
                  >
                    Food (Pasta)
                  </p>
                </div>
              </div>
              <div
                className="sc-ftTHYK caKGMS"
              />
            </div>
          </div>
          <div
            className="sc-dkrFOg iqtpTI"
          />
          <div
            className="sc-dkrFOg gHMHIB"
          />
          <div
            className="sc-dkrFOg gHMHIB"
          />
          <div
            className="sc-dkrFOg iqtpTI"
          />
          <div
            className="sc-dkrFOg iKRSwo"
            onClick={[Function]}
          />
          <div
            className="sc-dkrFOg iKRSwo"
            onClick={[Function]}
          />
          <div
            className="sc-dkrFOg iKRSwo"
            onClick={[Function]}
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
